lib/Pinto/Schema/Result/Distribution.pm

```diff
@@ -220,15 +220,16 @@ sub FOREIGNBUILDARGS {
 sub register {
     my ( $self, %args ) = @_;
 
-    my $stack = $args{stack};
-    my $force = $args{force} || 0;
-    my $pin   = $args{pin}   || 0;
+    my $stack   = $args{stack};
+    my $force   = $args{force}   || 0;
+    my $pin     = $args{pin}     || 0;
+    my $dry_run = $args{dry_run} || 0;
 
     my $can_intermingle = $stack->repo->config->intermingle;
     my $did_register    = 0;
 
     $stack->assert_is_open;
-    $stack->assert_not_locked;
+    $stack->assert_not_locked unless $dry_run;
```

Comment:
I wonder if this call to `assert_not_locked` can be removed (i.e. trust the caller) so that you don't have to pass `$dry_run` all over the place.